lxc/storage_volume.go

```diff
@@ -859,10 +859,15 @@ func (c *cmdStorageVolumeDetach) run(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	volName, volType := parseVolume("custom", args[1])
+	if volType != "custom" {
+		return errors.New(i18n.G(`Only "custom" volumes can be attached to instances`))
```

Comment:
@MggMuggins is this going to need to change in your PR that adds support for attaching root disks to instances?

Reply:
Yup, he has taken a look at these changes. I believe these checks will also consider `virtual-machine` volumes.

Reply:
Yes, this should conflict with my PR; I'm happy to rebase.